apps/builder/app/dashboard/loading-state/loading.tsx

```diff
@@ -0,0 +1,43 @@
+import { useEffect, useState } from "react";
+import { Flex, css } from "@webstudio-is/design-system";
+import spinner from "./spinner.svg";
+
+const containerStyle = css({
+  position: "absolute",
+  left: "0",
+  right: "0",
+  top: "0",
+  bottom: "0",
+  background:
+    "radial-gradient(34.37% 50% at 50% 50%, rgba(255, 255, 255, 0.5) 0%, rgba(248, 248, 248, 0.5) 100%);",
+  backdropFilter: "blur(8px)",
+  display: "flex",
+  alignItems: "center",
+  justifyContent: "center",
+});
+
+export const Loading = ({ delay = 0 }: { delay?: number }) => {
+  const [isVisible, setIsVisible] = useState(false);
+
+  useEffect(() => {
+    // we're using a timeout to avoid showing the loading state for a very short time
+    // If the connection is very fast, the user will not see a loading state
+    // This is just an idea, we can remove it if we don't like it
+    const timer = setTimeout(() => {
+      setIsVisible(true);
+    }, delay);
+
+    return () => clearTimeout(timer);
+  }, [delay]);
+
+  return isVisible ? (
+    <Flex
+      direction="row"
+      justify="center"
+      align="center"
+      className={containerStyle()}
+    >
+      <img src={spinner} alt="Loading" width={64} height={64} />
```

Comment:
in this case it's loading svg over http, could be also a good option, @TrySound wdyt over http or inline string?

Reply:
Depends but if svg is huge user won't see spinner until it's loaded and by then page may be ready.

Reply:
2kb is huge?

Reply:
let go with http-based I guess, it's even 1.5kb

Reply:
Maybe not that bad if add image to prefetch

Reply:
actually, the problem can be if user lost internet, we won't be able to show indicator, so either prefetch or inline, I would go with inline, because adding prefetch is additional work